include/eve/detail/function/simd/x86/load.hpp

```diff
@@ -72,4 +86,36 @@ namespace eve::detail
       }
     }
   }
+
+  //================================================================================================
+  // logical loads require some specific setup
+  //================================================================================================
+  template<typename T, typename N, typename Ptr, x86_abi ABI>
+  EVE_FORCEINLINE
+  auto load(eve::as_<logical<wide<T, N, ABI>>> const &, ABI const & abi, Ptr p)
+  requires( std::same_as<logical<T>, std::remove_cvref_t<decltype(*p)>> )
+  {
+    auto block = [&]() -> wide<T, N, ABI>
+    {
+      using tgt = eve::as_<wide<T, N, ABI>>;
+      if constexpr( !std::is_pointer_v<Ptr> ) return load(tgt(), abi, (T const*)(p.get()));
+      else                                    return load(tgt(), abi, (T const*)(p));
+    }();
+
+    return is_nez(block).storage();
+  }
+
+  template<typename Iterator, typename T, typename N, x86_abi ABI>
+  EVE_FORCEINLINE auto load ( eve::as_<logical<wide<T, N, ABI>>> const &
+                            , ABI const &, Iterator b, Iterator e
+                            ) noexcept
+  {
+    auto block = [&]() -> wide<T, N, ABI>
+    {
+      using tgt = eve::as_<wide<T, N, ABI>>;
+      return load(tgt(), ABI{}, b, e);
+    }();
+
+    return is_nez(block).storage();
```

Comment:
I am very confused about this `is_nez`.
I also think that you don'y return a logical but rather it's underlying type that can be convertible to logical.

Reply:
As per Slack discussion, this is required to load data from scalar logical then compress to mmask.
This is due to be replaced when the typeless masks are implemented.